### PR TITLE
Updates to hand functions

### PR DESCRIPTION
**Pull request by Affonso-Gui**
Open to discussions

### CODE REVIEW

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l [hunk 1]
```diff
@@ -303,36 +303,44 @@
...
-      (setq a1deg (rad2deg a1rad))
-      (float-vector a1deg (- a1deg) (- a1deg) a1deg)))
-  (:set-hand-width (hand width &key (tm 1000) effort)
-    (when effort
-      (send self :hand-effort hand effort))
-    (send self :hand-angle-vector hand (send self :hand-width2angles width) tm))
+      (let ((a (rad2deg %(pi/2 - acos((width / 2.0 + safetymargin - w0) / l1)))))
+        (float-vector a (- a) (- a) a))))
+  (:hand-angles2width (vec)
+    (assert (= (length vec) 4))
```
Comment (pazeshun) [inline hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: Can you add printing error message here?

**Review by pazeshun — CHANGES_REQUESTED** [review]

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l (hunk 1 above)
Comment (Affonso-Gui) [reply hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: Added

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l [hunk 2]
```diff
@@ -283,60 +307,77 @@
...
+      (cond
+       ((numberp effort)
+        ;; setmaxtorque with same effort value
         (mapcar
          #'(lambda (id) (send self :call-operation-return :servocontrollerservice_setmaxtorque :id id :percentage effort))
-         ids)
-        ;; getmaxtorque
+         ids))
+       ;; setmaxtorque with different effort values
+       (effort
```
Comment (pazeshun) [inline hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: `(listp effort)` or `(vectorp effort)`?

Comment (pazeshun) [reply hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: You need to check length of `effort`.

**Review by pazeshun — CHANGES_REQUESTED** [review]

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l (hunk 2 above)
Comment (Affonso-Gui) [reply hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: Fixed it and added more typecheck to surrounding functions

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l [hunk 3]
```diff
@@ -251,13 +254,41 @@
...
 (defmethod hironxjsk-interface
-  (:hand-angle-vector
-   (&optional (hand :hands) av (tm 1000))
+  (:hand-angle-vector (hand &optional av (tm 1000))
+   ;; check type
+   (when av
+     (case hand
+           (:hands
+            (if (= (length av) +hironx-hand-servo-num+) (setq av (concatenate float-vector av av)))
+            (assert (= (length av) (* 2 +hironx-hand-servo-num+))))
```
Comment (pazeshun) [inline hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: Add printing info like https://github.com/pazeshun/rtmros_tutorials/pull/2/files#diff-6cffc2ae28c25e5e9c2a0377adcf4031R367

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l [hunk 4]
```diff
@@ -251,13 +254,41 @@
...
+           ((:rhand :lhand)
+            (assert (= (length av) +hironx-hand-servo-num+)))))
```
Comment (pazeshun) [inline hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: Add printing info like https://github.com/pazeshun/rtmros_tutorials/pull/2/files#diff-6cffc2ae28c25e5e9c2a0377adcf4031R367

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l [hunk 5]
```diff
@@ -283,60 +314,89 @@
...
+        (mapcar
+         #'(lambda (id) (send self :call-operation-return :servocontrollerservice_setmaxtorque :id id :percentage effort))
+         ids))
+       ((or (consp effort) (vectorp effort))
+        ;; check length
+        (case hand
+              (:hands
+               (if (= (length effort) +hironx-hand-servo-num+)
+                   (setq effort (concatenate float-vector effort effort)))
+               (assert (= (length effort) (* 2 +hironx-hand-servo-num+))))
```
Comment (pazeshun) [inline hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: Add printing info like https://github.com/pazeshun/rtmros_tutorials/pull/2/files#diff-6cffc2ae28c25e5e9c2a0377adcf4031R367

File: hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l [hunk 6]
```diff
@@ -283,60 +314,89 @@
...
+              ((:rhand :lhand)
+               (assert (= (length effort) +hironx-hand-servo-num+))))
```
Comment (pazeshun) [inline hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l]: Add printing info like https://github.com/pazeshun/rtmros_tutorials/pull/2/files#diff-6cffc2ae28c25e5e9c2a0377adcf4031R367

**Review by pazeshun — CHANGES_REQUESTED** [review]

**Review by pazeshun — APPROVED** [review]

**Comment by pazeshun** [comment]
I will merge this after checking in next demos.

**Comment by Affonso-Gui** [comment]
@pazeshun Forget to mention, but this will have better debugging printing information with https://github.com/euslisp/EusLisp/pull/363
https://github.com/pazeshun/rtmros_tutorials/blob/4ddedec446628ce74dc146641e01028d1f5f0aad/hrpsys_ros_bridge_tutorials/euslisp/hironxjsk-interface.l#L259-L260

**Pull request merged**